src/game_objects/grid.cpp

```diff
@@ -1,11 +1,71 @@
 #include "grid.h"
 #include "events.h"
 #include "game/phase.h"
+#include "util/colors.h"
 
-Grid::Grid(int w, int h) :
+#include <iostream>
+#include <fstream>
+
+// Default color is white, meaning that it only uses texture
+glm::vec3 default_color = color::white;
+// When selected, will tint the tile green, but keeps texture
+glm::vec3 select_color = color::green;
+
+Grid::Grid(const char *map_loc) :
         hover(nullptr) {
-    width = w;
-    height = h;
+
+    // Loads map from file
+    std::ifstream fin;
+    fin.open(map_loc);
```

Comment:
I put the parsing stuff in the constructor since I wasn't sure if there was a better place to put it. Let me know if anyone has a better suggestion.